concurrency.go

```diff
@@ -93,3 +96,35 @@ func Async6[A, B, C, D, E, F any](f func() (A, B, C, D, E, F)) <-chan Tuple6[A,
 	}()
 	return ch
 }
+
+// WaitFor runs periodically until a condition is validated.
+func WaitFor(condition func(i int) bool, maxDuration time.Duration, tick time.Duration) (int, time.Duration, bool) {
```

Comment:
What about this


```suggestion
func WaitFor(condition func(i int) bool, maxDuration time.Duration, tick time.Duration) (int, time.Duration, bool) {
       return WaitForWithCtx(context.Background(), condition, maxDuration, tick)
}


// WaitForWithCtx runs periodically until a condition is validated or context is cancelled
 func WaitForWithCtx(ctx context.Context, condition func(i int) bool, maxDuration time.Duration, tick time.Duration) (int, time.Duration, bool) { 
         if condition(0) { 
                 return 1, 0, true 
         } 
  
         start := time.Now() 
  
         timer := time.NewTimer(maxDuration) 
         ticker := time.NewTicker(tick) 
  
         defer func() { 
                 timer.Stop() 
                 ticker.Stop() 
         }() 
  
         i := 1 
  
         for { 
                 select { 
                 case <-ctx.Done():
                         return i, time.Since(start), false
                 case <-timer.C: 
                         return i, time.Since(start), false 
                 case <-ticker.C: 
                         if condition(i) { 
                                 return i + 1, time.Since(start), true 
                         } 
  
                         i++ 
                 } 
         } 
 }
```

Reply:
@samber

Reply:
Please open a new PR.

Reply:
I opened
-  #480